pandas/core/nanops.py

```diff
@@ -344,12 +344,22 @@ def nanmedian(values, axis=None, skipna=True):
 
     values, mask, dtype, dtype_max = _get_values(values, skipna)
 
+    if not skipna:
```

Comment:
yes, so use the ``skipna`` flag as a switch when generating which function to use.